File: BIASecurity/BIASecurity/BIASecurity.WebAPI/Model/EAACIP.cs
using BIACore.Model;
using System;
using System.Collections.Generic;
using System.Data;

using extjs = BIACore.Web.Model.ExtJS;

namespace BIASecurity.WebAPI.Model
{
    public class EAACIP : extjs.Parameter
    {
        //public int DBNullValue = DBNull.Value;
        public string BIA_ID { get; set; }
        //public string CSC_Access { get; set; }
       //// public string CSC_Type_Country_Code { get; set; }

        /// <summary>
        /// ///
        
        public string Region_Coordinator { get; set; }
        public string District_Coordinator { get; set; }
        public string Add_Address { get; set; }
        public int Read_Only_Access { get; set; }
        public int View_AddrKey { get; set; }
        public int Change_ActiveFlag { get; set; }
        public int Unclassified_Des { get; set; }
        public int Edit_Reasons { get; set; }
        public string CSC_Access { get; set; }
        public int CSC_Supervisor_Access { get; set; }
        public int Customer_Contact_Export { get; set; }
        public int View_Repeater_Rpt { get; set; }
        public int EA_ACIP_View_Usage_Rpt { get; set; }
        public int Escalation_Export { get; set; }
        public int EA_UseMultiTrackingNumLookup { get; set; }
        public int EA_Weekly_Designation_Summary_Report { get; set; }
        public string EA_ACIP_Online_Research_Agent { get; set; }
        public string EA_ACIP_Online_Research_Agent_Reports { get; set; }
        public string EA_ACIP_ORA_Level { get; set; }
        public int EA_ACIP_Can_Process_Multi_Tenant { get; set; } 
        public int EA_ACIP_Can_Designate { get; set; }
        public int EA_ACIP_Can_View_AMS_30Day_Audit { get; set; }

        /// </summary>
        /// <returns></returns>
        public override DBParameter[] ToDBParameter()
        {
            List<DBParameter> args = new List<DBParameter>();
            if (!string.IsNullOrEmpty(BIA_ID)) args.Add(new DBParameter("@BIA_ID", DbType.AnsiString, BIA_ID));
            //if (!string.IsNullOrEmpty(CSC_Access)) args.Add(new DBParameter("@CSC_Type", DbType.AnsiString, CSC_Access));
            //if (!string.IsNullOrEmpty(CSC_Type_Country_Code)) args.Add(new DBParameter("@CSC_Type_Country_Code", DbType.AnsiString, CSC_Type_Country_Code));

            if (!string.IsNullOrEmpty(Region_Coordinator)) args.Add(new DBParameter("@Region_Coordinator", DbType.AnsiString, Region_Coordinator));
            if (!string.IsNullOrEmpty(District_Coordinator)) args.Add(new DBParameter("@District_Coordinator", DbType.AnsiString, District_Coordinator));
            if (!string.IsNullOrEmpty(Add_Address)) args.Add(new DBParameter("@Add_Address", DbType.AnsiString, Add_Address));
                args.Add(new DBParameter("@Read_Only_Access", DbType.Int32, Read_Only_Access));
            //if (!string.IsNullOrEmpty(Read_Only_Access)) args.Add(new DBParameter("@Read_Only_Access", DbType.Int32, Read_Only_Access));
              //  else args.Add(new DBParameter("@Read_Only_Access", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@View_AddrKey", DbType.Int32, View_AddrKey));
                args.Add(new DBParameter("@Change_ActiveFlag", DbType.Int32, Change_ActiveFlag));
                args.Add(new DBParameter("@Unclassified_Des", DbType.Int32, Unclassified_Des));
                //else args.Add(new DBParameter("@Unclassified_Des", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@Edit_Reasons", DbType.Int32, Edit_Reasons));
            if (!string.IsNullOrEmpty(CSC_Access)) args.Add(new DBParameter("@CSC_Access", DbType.AnsiString, CSC_Access));
                else args.Add(new DBParameter("@Edit_Reasons", DbType.AnsiString, DBNull.Value));
                args.Add(new DBParameter("@CSC_Supervisor_Access", DbType.Int32, CSC_Supervisor_Access));
                //else args.Add(new DBParameter("@CSC_Supervisor_Access", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@Customer_Contact_Export", DbType.Int32, Customer_Contact_Export));
               // else args.Add(new DBParameter("@Customer_Contact_Export", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@View_Repeater_Rpt", DbType.Int32, View_Repeater_Rpt));
                //else args.Add(new DBParameter("@View_Repeater_Rpt", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@EA_ACIP_View_Usage_Rpt", DbType.Int32, EA_ACIP_View_Usage_Rpt));
               // else args.Add(new DBParameter("@EA_ACIP_View_Usage_Rpt", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@Escalation_Export", DbType.Int32, Escalation_Export));
                //else args.Add(new DBParameter("@Escalation_Export", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@EA_UseMultiTrackingNumLookup", DbType.Int32, EA_UseMultiTrackingNumLookup));
                //else args.Add(new DBParameter("@EA_UseMultiTrackingNumLookup", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@EA_Weekly_Designation_Summary_Report", DbType.Int32, EA_Weekly_Designation_Summary_Report));
                //else args.Add(new DBParameter("@EA_Weekly_Designation_Summary_Report", DbType.AnsiString, DBNullValue));             
            if (!string.IsNullOrEmpty(EA_ACIP_Online_Research_Agent)) args.Add(new DBParameter("@EA_ACIP_Online_Research_Agent", DbType.AnsiString, EA_ACIP_Online_Research_Agent));
                else args.Add(new DBParameter("@EA_ACIP_Online_Research_Agent", DbType.AnsiString, DBNull.Value));
            if (!string.IsNullOrEmpty(EA_ACIP_Online_Research_Agent_Reports)) args.Add(new DBParameter("@EA_ACIP_Online_Research_Agent_Reports", DbType.AnsiString, EA_ACIP_Online_Research_Agent_Reports));
                else args.Add(new DBParameter("@EA_ACIP_Online_Research_Agent_Reports", DbType.AnsiString, DBNull.Value));
            if (!string.IsNullOrEmpty(EA_ACIP_ORA_Level)) args.Add(new DBParameter("@EA_ACIP_ORA_Level", DbType.AnsiString, EA_ACIP_ORA_Level));
                else args.Add(new DBParameter("@EA_ACIP_ORA_Level", DbType.AnsiString, DBNull.Value));
                args.Add(new DBParameter("@EA_ACIP_Can_Process_Multi_Tenant", DbType.Int32, EA_ACIP_Can_Process_Multi_Tenant));
                //else args.Add(new DBParameter("@EA_ACIP_Can_Process_Multi_Tenant", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@EA_ACIP_Can_Designate", DbType.Int32, EA_ACIP_Can_Designate));
                //else args.Add(new DBParameter("@EA_ACIP_Can_Designate", DbType.AnsiString, DBNullValue));
                args.Add(new DBParameter("@EA_ACIP_Can_View_AMS_30Day_Audit", DbType.Int32, EA_ACIP_Can_View_AMS_30Day_Audit));
               // else args.Add(new DBParameter("@EA_ACIP_Can_View_AMS_30Day_Audit", DbType.AnsiString, DBNullValue));



            args.AddRange(base.ToDBParameter());
            return args.ToArray();
        }
    }
}

File: BIASecurity/BIASecurity/BIASecurity.WebSite/App/View/EA/EAACIP.js
Ext.define('App.View.EA.EAACIP', {
    extend: 'Ext.container.Container',
    alias: 'widget.App-View-EA-EAACIP',
    itemId: 'EAACIPPanelId',
    closable: true,
    hidden: true,
    scrollable: false,
    minWidth: 500,
    minHeight: 500,
    bodyPadding: 10,
    collapsible: true,
    defaults: {
        margin: '5 2 5 2',
        padding: '8 5 8 5',
        border: false,
    },
    layout: {
        type: 'hbox',
        align: 'middle',
        pack: 'center'
    },
    viewModel: {
        data:
        {
            "Region_Coordinator": "No",
            "District_Coordinator": "No",
            "Add_Address": "No",
            "Read_Only_Access": "No",
            "View_AddrKey": "No",
            "Change_ActiveFlag": "No",
            "Unclassified_Des": "No",
            "Edit_Reasons": "No",
            "CSC_Access": "No",
            "CSC_Supervisor_Access": "No",
            "Customer_Contact_Export": "No",
            "View_Repeater_Rpt": "No",
            "EA_ACIP_View_Usage_Rpt": "No",
            "Escalation_Export": "No",
            "EA_UseMultiTrackingNumLookup": "No",
            "EA_Weekly_Designation_Summary_Report": "No",
            "EA_ACIP_Online_Research_Agent": "No",
            "EA_ACIP_Online_Research_Agent_Reports": "No",
            "EA_ACIP_ORA_Level": "No",
            "EA_ACIP_Can_Process_Multi_Tenant": "No",
            "EA_ACIP_Can_Designate": "No",
            "EA_ACIP_Can_View_AMS_30Day_Audit": "No"
        }

    },   
    items: [                
            {
            xtype: 'form',
            border: false,
            itemId: 'EAACIPContentTable',
            layout: {
                type: 'table',
                columns: 3,
                align: 'stretch'
            },
            defaults: {
                margin: '2 2 2 2',
                padding: '2 2 2 2',
                labelWidth: 150,
                width: 350
            },
            buttonAlign: 'center',
            buttons: [
                { xtype: 'button', itemId: 'btnEASave', text: 'Save', margin: '5 5 5 5' },
                { xtype: 'button', itemId: 'btnEACancel', text: 'Cancel', margin: '5 5 5 5' }
            ],

            items: [                    
                    {
                        xtype: 'combobox',
                        fieldLabel: 'Region Coordinator :',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'Region Coordinator :',
                         itemId: 'acipRegionCoordinatorId',
                         bind: '{Region_Coordinator}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": "Yes" },
                                { "Name": "No", "Code": "No" }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                    {
                        xtype: 'combobox',
                        fieldLabel: 'District Coordinator :',
                        tooltip: 'District Coordinator :',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        itemId: 'acipDistrictCoordinatorId',
                        bind: '{District_Coordinator}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": "Yes" },
                                { "Name": "No", "Code": "No" }
                            ]
                        },
                        queryMode: 'local',                            
                        typeAhead: true
                    },
                    {
                        xtype: 'combobox',
                        fieldLabel: 'Can Update Addresses:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'Test address',//These users will see the Quicksearch link in ACIP.  The link allows them to update addresses.
                        itemId: 'acipAddAddressId',
                        bind: '{Add_Address}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": "Yes" },
                                { "Name": "No", "Code": "No" }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'ReadOnly Access:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'Test address',//<td title="These users can only view the designated report." class="Label">ReadOnly Access:</td>
                        itemId: 'acipReadOnlyAccessId',
                        bind: '{Read_Only_Access}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Country:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'CSC_Type_Country_Code',
                        valueField: 'CSC_Type_Country_Code',
                        tooltip: 'Select CSC Country.',
                        itemId: 'acipCountryId',
                        store: {
                            fields: ['CSC_Type_Country_Code'],
                            type: 'webapi',
                            storeId: 'acipCountryStore',
                            autoLoad: true,
                            api: {
                                read: 'api/ACIP/GetCSCCountryCodes'
                            }
                            },  
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'CSC User Location:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'CSC_Type_Desc',
                        valueField: 'CSC_Type',
                        tooltip: 'These users can search for and dispute addresses',
                        itemId: 'acipCSCAccessId',
                        bind: '{CSC_Access}',
                        store: {
                            fields: ['CSC_Type', 'CSC_Type_Desc'],
                            type: 'webapi',
                            storeId:'acipCSCAccessStore',
                            autoLoad: false,
                            api: {
                                read: 'api/ACIP/GetCSCAccessLevels'
                            }
                        },                         
                        queryMode: 'local',
                        typeAhead: true
                    },
                    {
                        xtype: 'combobox',
                        fieldLabel: 'CSC Supervisor Access:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'The CSC Supervisor can override confirmed address designations.',//<td title="These users can only view the designated report." class="Label">ReadOnly Access:</td>
                        itemId: 'acipCSCSupervisorAccessId',
                        bind: '{CSC_Supervisor_Access}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Edit Designation Reasons:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'These users have the ability to edit designation reason options.',//<td title="These users can only view the designated report." class="Label">ReadOnly Access:</td>
                        itemId: 'acipEditReasonsId',
                        bind: '{Edit_Reasons}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Unclassified Designations:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'These users can classify addresses as Unclassified (Region Coordinators Only).',
                        itemId: 'acipUnclassifiedDesId',
                        bind: '{Unclassified_Des}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'View Repeater Report:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'These users can view the Repeater report',
                        itemId: 'acipViewRepeaterRptId',
                        bind: '{View_Repeater_Rpt}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Customer Contact Export:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'These users can export the latest address disputes',
                        itemId: 'acipCustomerContactExportId',
                        bind: '{Customer_Contact_Export}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'View AddrKey:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'Developers only.',
                        itemId: 'acipViewAddrKeyId',
                        bind: '{View_AddrKey}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'View Usage Report:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'Developers only.',
                        itemId: 'acipEAACIPViewUsageRptId',
                        bind: '{EA_ACIP_View_Usage_Rpt}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'View Escalation Report:',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'View Escalation Report.',
                        itemId: 'acipEscalationExportId',
                        bind: '{Escalation_Export}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Use Multi Tracking Number Lookup',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'Use Multi Tracking Number Lookup.',
                        itemId: 'acipEAUseMultiTrackingNumLookupId',
                        bind: '{EA_UseMultiTrackingNumLookup}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'View Weekly Designation Summary Report',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'View Weekly Designation Summary Report.',
                        itemId: 'acipEAWeeklyDesignationSummaryReportId',
                        bind: '{EA_Weekly_Designation_Summary_Report}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                    //  {
                    //    xtype: 'combobox',
                    //    fieldLabel: 'ORA Level',
                    //    labelWidth: 150,
                    //    grow: true,
                    //    growToLongestValue: true,
                    //    displayField: 'EA_ACIP_ORA_Level',
                    //    valueField: 'EA_ACIP_ORA_Level',
                    //    tooltip: 'EA_ACIP_ORA_Level.',
                    //    itemId: 'acipEAACIPORALevelId',
                    //    store: {
                    //        fields: ['EA_ACIP_ORA_Level'],
                    //        data: [
                    //            { "EA_ACIP_ORA_Level": "Corporate CSC"  },
                    //            { "EA_ACIP_ORA_Level": "Manager"  },
                    //            { "EA_ACIP_ORA_Level": "Supervisor"  },
                    //            { "EA_ACIP_ORA_Level": "Coach"  },
                    //            { "EA_ACIP_ORA_Level": "Agent"  }
                    //        ]
                    //    },
                    //    queryMode: 'local',
                    //    typeAhead: true
                    //},
                    {
                        xtype: 'combobox',
                        fieldLabel: 'Can Process Multi-Tenant Investigations',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'EA_ACIP_Can_Process_Multi_Tenant.',
                        itemId: 'acipEAACIPCanProcessMultiTenantId',
                        bind: '{EA_ACIP_Can_Process_Multi_Tenant}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Can Designate',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'EA_ACIP_Can_Designate.',
                        itemId: 'acipEAACIPCanDesignateId',
                        bind: '{EA_ACIP_Can_Designate}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },
                    {
                        xtype: 'combobox',
                        fieldLabel: 'ORA Level',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'EA_ACIP_ORA_Level',
                        valueField: 'EA_ACIP_ORA_Level',
                        tooltip: 'EA_ACIP_ORA_Level.',
                        itemId: 'acipEAACIPORALevelId',
                        bind: '{EA_ACIP_ORA_Level}',
                        store: {
                            fields: ['EA_ACIP_ORA_Level'],
                            data: [
                                { "EA_ACIP_ORA_Level": "Corporate CSC" },
                                { "EA_ACIP_ORA_Level": "Manager" },
                                { "EA_ACIP_ORA_Level": "Supervisor" },
                                { "EA_ACIP_ORA_Level": "Coach" },
                                { "EA_ACIP_ORA_Level": "Agent" }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                    },//for testing of alignment. not actual order 
                        {
                        xtype: 'combobox',
                        fieldLabel: 'Can View AMS 30 Day Audit Report',
                        labelWidth: 150,
                        grow: true,
                        growToLongestValue: true,
                        displayField: 'Name',
                        valueField: 'Code',
                        tooltip: 'EA_ACIP_Can_View_AMS_30Day_Audit.',
                        itemId: 'acipEAACIPCanViewAMS30DayAuditId',
                        bind: '{EA_ACIP_Can_View_AMS_30Day_Audit}',
                        store: {
                            fields: ['Name', 'Code'],
                            data: [
                                { "Name": "Yes", "Code": 1 },
                                { "Name": "No", "Code": 0 }
                            ]
                        },
                        queryMode: 'local',
                        typeAhead: true
                },
                    {
                        xtype: 'combobox',
                        hidden:true,                       
                        displayField: 'CSC_Type_Country_Code',
                        valueField: 'CSC_Type_Country_Code',
                        itemId: 'acipCSCTypeCountryCodeId',
                        store: {
                            fields: ['CSC_Type_Country_Code'],
                            type: 'webapi',
                            autoLoad: false,
                            api: {
                                read: 'api/ACIP/GetCSCTypeCountryCode'
                            }
                        }
                    },
                    { xtype: 'hiddenfield', name: 'HAdd_Address', itemId: 'HAdd_Address' },
                    { xtype: 'hiddenfield', name: 'HUnclassified_Des', itemId: 'HUnclassified_Des' },
                    ///////////////////////////////

                ]
            }, //END of form fields            
     ]
        
});